pkg/v22/resource/app/desired.go

```diff
@@ -92,7 +92,14 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
 	var apps []*g8sv1alpha1.App
 
 	for _, appSpec := range r.newAppSpecs() {
-		userConfig := newUserConfig(clusterConfig, appSpec, configMaps, secrets)
+		userConfig, err := newUserConfig(clusterConfig, appSpec, configMaps, tenantConfigMaps, secrets)
+		if IsNotMigratedError(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q user values not migrated yet, continuing", appSpec.App))
+			continue
```

Comment:
This is the important bit so we don't create the app CR until the user values have been migrated.